### PR TITLE
add az vmware datastore command group

### PR DESCRIPTION
**Pull request by ctaggart**
Commands that use the API from Azure/azure-rest-api-specs#12879

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
vmware

File: src/vmware/azext_vmware/_help.py [hunk 1]
```diff
@@ -151,3 +156,23 @@
     type: command
     short-summary: Return trial status for subscription by region.
 """
+
+helps['vmware datastore create'] = """
+    type: command
+    short-summary: Create a datastore in a private cloud cluster.
```
Comment (yungezz) [inline src/vmware/azext_vmware/_help.py:162]: pls add examples to help

File: src/vmware/azext_vmware/_params.py [hunk 2]
```diff
@@ -70,3 +70,13 @@ def load_arguments(self, _):
 
     with self.argument_context('vmware hcx-enterprise-site') as c:
         c.argument('name', options_list=['--name', '-n'], help='The name of the HCX Enterprise Site.')
+
+    with self.argument_context('vmware datastore') as c:
+        c.argument('name', options_list=['--name', '-n'], help='The name of the datastore.')
+        c.argument('cluster', help='The name of the cluster.')
```
Comment (yungezz) [inline src/vmware/azext_vmware/_params.py:76]: does it support id?

File: src/vmware/azext_vmware/_params.py [hunk 3]
```diff
@@ -70,3 +70,13 @@ def load_arguments(self, _):
...
         c.argument('name', options_list=['--name', '-n'], help='The name of the HCX Enterprise Site.')
+
+    with self.argument_context('vmware datastore') as c:
+        c.argument('name', options_list=['--name', '-n'], help='The name of the datastore.')
+        c.argument('cluster', help='The name of the cluster.')
+
+    with self.argument_context('vmware datastore create') as c:
+        c.argument('nfs_provider_ip', help='IP address of the NFS provider.')
+        c.argument('nfs_file_path', help='File path through which the NFS volume is exposed by the provider.')
+        c.argument('endpoints', nargs='*', help='iSCSI provider target IP address list.')
```
Comment (yungezz) [inline src/vmware/azext_vmware/_params.py:81]: any validator?

File: src/vmware/azext_vmware/_params.py (hunk 2 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/_params.py:76]: It does not. I'm curious what you are recommending.

File: src/vmware/azext_vmware/_params.py (hunk 3 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/_params.py:81]: Not currently. I think it is a good idea. Can we add it into the next version?

File: src/vmware/azext_vmware/_help.py (hunk 1 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/_help.py:162]: @VamsiSamgam, can you add the requested examples?

**Comment by ctaggart** [comment]
@qwordy, after this pull request is merged, can you please publish 2.0.0 of the vmware extension. This is the final change.

I am going to be out for the rest of the week for medical reasons. @VamsiSangam & @Raaghavi both have contributor access to my fork and can complete any of the remaining minor adjustments if they are required.

File: src/vmware/azext_vmware/_help.py (hunk 1 above)
Comment (cjain85) [reply src/vmware/azext_vmware/_help.py:162]: @yungezz  examples have been added , Can you please take a look

Comment (Raaghavi) [reply src/vmware/azext_vmware/_help.py:162]: Examples added.

**Comment by ctaggart** [comment]
HI @yungezz & @qwordy, the examples have been added for datastore commands. Can we please get this merged and have the new extension version published?

**Comment by Raaghavi** [comment]
@qwordy & @yungezz Can we please have this PR reviewed?

File: src/vmware/azext_vmware/_params.py (hunk 2 above)
Comment (yungezz) [reply src/vmware/azext_vmware/_params.py:76]: cli convention is to support both resource id and resource name at one param, such as storage_account, to provide convinience for users

File: src/vmware/azext_vmware/_params.py (hunk 3 above)
Comment (yungezz) [reply src/vmware/azext_vmware/_params.py:81]: ok

File: src/vmware/azext_vmware/_params.py (hunk 2 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/_params.py:76]: I created https://github.com/Azure/azure-cli-extensions/issues/3020 and will add it to a 2.1 of vmware ext.

File: src/vmware/azext_vmware/_params.py (hunk 3 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/_params.py:81]: I opened https://github.com/Azure/azure-cli-extensions/issues/3021 and assigned it to myself.

**Review by yungezz — APPROVED** [review]

**Comment by qwordy** [comment]
OK. I am reviewing it.

File: src/vmware/azext_vmware/custom.py [hunk 4]
```diff
@@ -166,3 +166,25 @@ def hcxenterprisesite_show(cmd, client: AVSClient, resource_group_name, private_
...
+    datastore = Datastore()
+    if nfs_provider_ip is not None or nfs_file_path is not None:
+        datastore.net_app_volume = NetAppVolume(nfs_provider_ip=nfs_provider_ip, nfs_file_path=nfs_file_path)
+    if len(endpoints) > 0 or lun_name is not None:
+        datastore.disk_pool_volume = DiskPoolVolume(endpoints=endpoints, lun_name=lun_name)
+    return client.datastores.begin_create(resource_group_name=resource_group_name, private_cloud_name=private_cloud, cluster_name=cluster, datastore_name=name, datastore=datastore)
+
+
+def datastore_list(cmd, client: AVSClient, resource_group_name, private_cloud, cluster):
+    return client.datastores.list(resource_group_name=resource_group_name, private_cloud_name=private_cloud, cluster_name=cluster)
```
Comment (qwordy) [inline src/vmware/azext_vmware/custom.py:182]: Does it support listing datastores in all resource groups in a subscription? Most Azure CLI list commands support this usage.

File: src/vmware/azext_vmware/tests/latest/test_datastores_scenario.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import unittest
+
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer)
+from msrestazure.azure_exceptions import CloudError
+
+
+class VmwareDatastoresScenarioTest(ScenarioTest):
+    def setUp(self):
+        # https://vcrpy.readthedocs.io/en/latest/configuration.html#request-matching
+        self.vcr.match_on = ['scheme', 'method', 'path', 'query']  # not 'host', 'port'
```
Comment (qwordy) [inline src/vmware/azext_vmware/tests/latest/test_datastores_scenario.py:16]: Does default configuration fail?

**Review by qwordy — APPROVED** [review]

**Pull request merged**